addons/account/tests/test_account_lock_exception.py

```diff
@@ -394,3 +394,28 @@ def test_company_lock_date(self):
                 }])
 
                 sp.close()  # Rollback to ensure all subtests start in the same situation
+
+    def test_user_exception_remove_lock_date(self):
+        """
+        Test that an exception removing a lock date (instead of just decreasing it) works.
+        """
+        for lock_date_field, move_type in self.soft_lock_date_info:
+            with self.subTest(lock_date_field=lock_date_field, move_type=move_type), self.cr.savepoint() as sp:
+                move = self.init_invoice(move_type, invoice_date='2016-01-01', post=True, amounts=[1000.0], taxes=self.tax_sale_a)
+
+                # Lock the move
+                self.company[lock_date_field] = fields.Date.to_date('2020-01-01')
+                with self.assertRaises(UserError), self.cr.savepoint():
```

Comment:
I'm not sure about this savepoint, is it necessary?

I have the feeling that it's useless according to [that](https://github.com/odoo/odoo/blame/45d21b0eeee595c5c3b3fe186cec0a98d8f64e48/odoo/tests/common.py#L465).